lib/omniauth/strategies/open_id.rb

```diff
@@ -59,13 +60,21 @@ def start
         end
       end
 
-      def get_identifier
+      def openid_form
         f = OmniAuth::Form.new(:title => 'OpenID Authentication')
         f.label_field('OpenID Identifier', options.identifier_param)
         f.input_field('url', options.identifier_param)
         f.to_response
       end
 
+      def get_identifier
+        if options[:on_login].respond_to?(:call)
+          instance_eval &options[:on_login]
```

Comment:
`instance_eval` feels like a smell here. Should be `options[:on_login].call(env)` I believe.